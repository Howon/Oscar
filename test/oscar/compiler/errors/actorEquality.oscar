message testNow()


actor Tester() {

  receive = {
    | testNow() => {
        Println("Error");
      }
    | die() => {
      }
  }
}

def main() => unit = {
  actor<Tester> tester1 = spawn actor<Tester>();
  actor<Tester> tester2 = spawn actor<Tester>();
  bool b = (tester1 == tester2);
  message<testNow>() |> tester1;
  message<die>() |> tester1;
  message<die>() |> tester2;
}
