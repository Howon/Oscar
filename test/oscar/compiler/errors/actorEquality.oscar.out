Error: Actors cannot be compared for equality
